### PR TITLE
Hotfix again

### PR DESCRIPTION
**Pull request by lance6716**
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `dm/dm-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: tests/sequence_sharding/data/db1.increment.sql [hunk 1]
```diff
@@ -3,7 +3,8 @@ insert into t1 (uid,name) values (100003,'NR');
 update t1 set name = 'uxoKehvqWg' where `uid` = 100001;
 update t1 set name = 'bapYymrtfT' where name = 'igvApUx';
 insert into t2 (uid,name) values (200004,'CXDvoltoliUINgo'),(200005,'188689130');
-alter table t1 add column c int;
+alter table T1 add column c int;
```
Comment (lance6716) [inline tests/sequence_sharding/data/db1.increment.sql:6]: I have checked without code changing, this test will fail and `query-status` shows similar result as we expected

**Comment by lance6716** [comment]
ptal @GMHDBJD @glorv

File: syncer/sharding_group.go [hunk 2]
```diff
@@ -380,6 +381,13 @@ func (sg *ShardingGroup) FlushData(targetTableID string) ([]string, [][]interfac
 
 // GenTableID generates table ID
 func GenTableID(schema, table string) (ID string, isSchemaOnly bool) {
+	log.L().Warn("hotfix, will change schema and table name to lower",
```
Comment (glorv) [inline syncer/sharding_group.go:384]: shall we only output this warn log when the lower value is different from the origin value?

Comment (lance6716) [reply syncer/sharding_group.go]: https://github.com/pingcap/dm/pull/1641/commits/99de7c5b600dac1909b4e9f2831c3f6ac622defe

File: syncer/sharding_group.go [hunk 3]
```diff
@@ -379,7 +380,28 @@ func (sg *ShardingGroup) FlushData(targetTableID string) ([]string, [][]interfac
 }
 
 // GenTableID generates table ID
-func GenTableID(schema, table string) (ID string, isSchemaOnly bool) {
+func GenTableID(schema, table string, upperSchema, upperTable []string) (ID string, isSchemaOnly bool) {
+	for _, s := range upperSchema {
+		if s == schema {
```
Comment (GMHDBJD) [inline syncer/sharding_group.go:385]: Maybe map will be better

**Review by GMHDBJD — COMMENTED** [review]
/lgtm

File: syncer/sharding_group.go [hunk 4]
```diff
@@ -379,7 +380,28 @@ func (sg *ShardingGroup) FlushData(targetTableID string) ([]string, [][]interfac
...
+			oldSchema := schema
+			schema = strings.ToLower(schema)
+			log.L().Warn("hotfix, changing schema to lowercase",
```
Comment (glorv) [inline syncer/sharding_group.go:388]:
```suggestion
			if oldSchema != schema {
			    log.L().Warn("hotfix, changing schema to lowercase",
			}
```

**Review by glorv — COMMENTED** [review]
rest LGTM

**Comment by lance6716** [comment]
manually test:

0. reset master in two upstreams, so binlog position is 154
1. delete `upper-table` in dm-task.yaml
2. run the case sequence_sharding
3. failed, stop-task
4. using this task
    ```
    ---
    name: sequence_sharding
    task-mode: incremental
    is-sharding: true
    meta-schema: "dm_meta"
    remove-meta: true
    enable-heartbeat: true
    timezone: "Asia/Shanghai"
    upper-table: ["T1"]

    target-database:
    host: "127.0.0.1"
    port: 4000
    user: "root"
    password: ""

    mysql-instances:
    - source-id: "mysql-replica-01"
        black-white-list:  "instance"
        route-rules: ["sharding-route-rules-table", "sharding-route-rules-schema"]
        column-mapping-rules: ["instance-1"]
        mydumper-config-name: "global"
        loader-config-name: "global"
        syncer-config-name: "global"
        meta:
        binlog-name: mysql-bin.000001
        binlog-pos: 154

    - source-id: "mysql-replica-02"
        black-white-list:  "instance"
        route-rules: ["sharding-route-rules-table", "sharding-route-rules-schema"]
        column-mapping-rules: ["instance-2"]
        mydumper-config-name: "global"
        loader-config-name: "global"
        syncer-config-name: "global"
        meta:
        binlog-name: mysql-bin.000001
        binlog-pos: 154

    black-white-list:
    instance:
        do-dbs: ["sharding_seq"]
        do-tables:
        -  db-name: "sharding_seq"
        tbl-name: "~^t[\\d]+"

    routes:
    sharding-route-rules-table:
        schema-pattern: sharding_seq*
        table-pattern: t*
        target-schema: sharding_target2
        target-table: t_target

    sharding-route-rules-schema:
        schema-pattern: sharding_seq*
        target-schema: sharding_target2

    column-mappings:
    instance-1:
        schema-pattern: "sharding_seq*"
        table-pattern: "t*"
        expression: "partition id"
        source-column: "id"
        target-column: "id"
        arguments: ["1", "", "t"]

    instance-2:
        schema-pattern: "sharding_seq*"
        table-pattern: "t*"
        expression: "partition id"
        source-column: "id"
        target-column: "id"
        arguments: ["2", "", "t"]

    mydumpers:
    global:
        mydumper-path: "./bin/mydumper"
        threads: 4
        chunk-filesize: 64
        skip-tz-utc: true
        extra-args: ""

    loaders:
    global:
        pool-size: 16
        dir: "./dumped_data"

    syncers:
    global:
        worker-count: 16
        batch: 100
        safe-mode: true
    ```
5. start-task
6. insert a row at upstream, chek that it's synced to downstream

File: syncer/sharding_group.go (hunk 3 above)
Comment (lance6716) [reply syncer/sharding_group.go:385]: for a small count, iter through array is faster than map (and i'm lazy)

**Pull request merged**